### PR TITLE
PIM-5277: Translate datetimepicker

### PR DESCRIPTION
**Pull request by pierallard**
| Q                 | A
| ----------------- | ---
| Specs             | n
| Behats            | y
| Blue CI           | TODO
| Changelog updated | ?
| Review and 2 GTM  | TODO
| Micro Demo (PO)   | y
| Migration script  | n
| Tech Doc          | n

This PR translates the datepicker. The translations will be done in Crowdin.
The JS part just set a configuration option to add the current locale translations, like described here.
http://www.malot.fr/bootstrap-datetimepicker/

The french translations are for demo-PO and behats.

### CODE REVIEW

File: src/Pim/Bundle/UIBundle/Resources/public/js/pim-datepicker.js [hunk 1]
```diff
@@ -1,18 +1,32 @@
...
                 options = $.extend(true, this.options, options);
 
+                if (('en' !== options.language) && (undefined === $.fn.datetimepicker.dates[options.language])) {
+                    var languageOptions = {};
+                    var defaultOptions = $.fn.datetimepicker.dates['en'];
+
+                    _.each(_.keys(defaultOptions), function (key) {
+                        languageOptions[key] = [];
+                        _.each(defaultOptions[key], function (value) {
+                            languageOptions[key].push(_.__('datetimepicker.'+key+'.'+value));
```
Comment (ClementGautier) [inline src/Pim/Bundle/UIBundle/Resources/public/js/pim-datepicker.js]: Use __ directly, if you use the underscore mixin you should not have to inject __.

File: features/Context/WebUser.php [hunk 2]
```diff
@@ -869,6 +869,18 @@ public function iChangeTheTo($field, $value = null, $language = null)
     }
 
     /**
+     * @param $field
+     *
+     * @When /^I click on (?P<field>\w+)$/
+     * @When /^I click on "(?P<field>[^"]+)"$/
+     */
+    public function iClickOnTheField($field)
```
Comment (jjanvier) [inline features/Context/WebUser.php]: nothing like already exists? nor in mink, nor in the pim?

Comment (pierallard) [inline features/Context/WebUser.php]: @jjanvier I searched for a similar function yesterday and ask in slack, apparently no.

File: features/Context/WebUser.php [hunk 3]
```diff
@@ -869,6 +869,18 @@ public function iChangeTheTo($field, $value = null, $language = null)
     }
 
     /**
+     * @param $field
+     *
+     * @When /^I click on (?P<field>\w+)$/
+     * @When /^I click on "(?P<field>[^"]+)"$/
```
Comment (ClementGautier) [inline features/Context/WebUser.php]: You can keep only one definition I guess. I would specify that you are clicking on a field too, like "I click on the field "toto""

File: src/Pim/Bundle/UIBundle/Resources/public/js/pim-datepicker.js [hunk 4]
```diff
@@ -1,18 +1,32 @@
...
                 options = $.extend(true, this.options, options);
 
+                if (('en' !== options.language) && (undefined === $.fn.datetimepicker.dates[options.language])) {
+                    var languageOptions = {};
+                    var defaultOptions = $.fn.datetimepicker.dates.en;
+
+                    _.each(_.keys(defaultOptions), function (key) {
+                        languageOptions[key] = [];
+                        _.each(defaultOptions[key], function (value) {
+                            languageOptions[key].push(_.__('datetimepicker.' + key + '.' + value));
```
Comment (ClementGautier) [inline src/Pim/Bundle/UIBundle/Resources/public/js/pim-datepicker.js]: as we said ... use __ ;)

**Comment by ClementGautier** [comment]
GTM after comments fixed

File: src/Pim/Bundle/DataGridBundle/Resources/public/js/datafilter/filter/date-filter.js [hunk 5]
```diff
@@ -27,9 +27,9 @@ function($, _, __, ChoiceFilter, Datepicker) {
                     '</select>' +
                 '</div>' +
                 '<div>' +
-                    '<span class="start"><input type="text" value="" class="<%= inputClass %> add-on" name="start" placeholder="from"></span>' +
+                    '<span class="start"><input type="text" value="" class="<%= inputClass %> add-on" name="start" placeholder="'+__('from')+'"></span>' +
                     '<span class="filter-separator">-</span>' +
-                    '<span class="end"><input type="text" value="" class="<%= inputClass %> add-on" name="end" placeholder="to"></span>' +
+                    '<span class="end"><input type="text" value="" class="<%= inputClass %> add-on" name="end" placeholder="'+__('to')+'"></span>' +
```
Comment (skeleton) [inline src/Pim/Bundle/DataGridBundle/Resources/public/js/datafilter/filter/date-filter.js]: add space between `+`
and better, create a template to add this html code

**Comment by solivier** [comment]
gtm

File: src/Pim/Bundle/UIBundle/Resources/translations/jsmessages.fr.yml [hunk 6]
```diff
@@ -45,3 +45,54 @@ jstree:
...
+    Su: Di
+    Mo: Lu
+    Tu: Ma
+    We: Me
+    Th: Je
+    Fr: Ve
+    Sa: Sa
+  months:
+    January: Janvier
+    February: Févier
```
Comment (skeleton) [inline src/Pim/Bundle/UIBundle/Resources/translations/jsmessages.fr.yml]: Févier :)

**Comment by skeleton** [comment]
gtm once fixed

File: src/Pim/Bundle/UIBundle/Resources/translations/jsmessages.fr.yml (hunk 6 above)
Comment (pierallard) [inline src/Pim/Bundle/UIBundle/Resources/translations/jsmessages.fr.yml]: The :eyes:  of the :tiger:

**Pull request merged**